app/lib/features/events/providers/events_provider.dart

```diff
@@ -0,0 +1,59 @@
+import 'dart:async';
+
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart'
+    show CalendarEvent, Space;
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class AsyncSpaceEventsNotifier
+    extends AutoDisposeFamilyAsyncNotifier<List<CalendarEvent>, Space> {
+  late Stream<void> _listener;
+  Future<List<CalendarEvent>> _getEvents() async {
+    return (await arg.calendarEvents()).toList(); // this might throw internally
+  }
+
+  @override
+  Future<List<CalendarEvent>> build(Space arg) async {
+    final client = ref.watch(clientProvider)!;
+    final spaceId = arg.getRoomId();
+    _listener = client.subscribe('$spaceId::CALENDAR'); // stay up to date
```

Comment:
aren't we doing lowercase keys? [Yeah, we are](https://github.com/acterglobal/a3/blob/main/native/core/src/statics.rs#L11).

```suggestion
    _listener = client.subscribe('$spaceId::calendar'); // stay up to date
```

Reply:
Sure, will replace it.